### PR TITLE
Provide offset option for showroom.

### PR DESCRIPTION
**Pull request by deiferni**
The offset is added to the current number and allows to display correct item number for batched items.

Closes https://github.com/4teamwork/ftw.showroom/issues/28
Fixes https://github.com/4teamwork/ftw.showroom/issues/30

### CODE REVIEW

File: ftw/showroom/js/src/register.js [hunk 1]
```diff
@@ -27,12 +27,16 @@ export default function Register(items = [], options) {
...
-      if (pointer === reveal.size - 1) {
-        options.tail(reveal.current);
-      }
+      performCalls();
+    }
+  }
+
+  function performCalls() {
+    if (pointer === 0) {
+      options.head(reveal.current);
```
Comment (bierik) [inline ftw/showroom/js/src/register.js]: I would prefer calling `current()` instead of `reveal.current`

File: ftw/showroom/js/src/register.js [hunk 2]
```diff
@@ -27,12 +27,16 @@ export default function Register(items = [], options) {
...
+    }
+    if (pointer === reveal.size - 1) {
+      options.tail(reveal.current);
```
Comment (bierik) [inline ftw/showroom/js/src/register.js]: I would prefer calling current() instead of reveal.current

File: ftw/showroom/js/src/showroom.js [hunk 3]
```diff
@@ -71,12 +72,11 @@ module.exports = function Showroom(items = [], options) {
...
-      $("#ftw-showroom-next", element).hide();
-    }
-    if(!register.hasPrev()) {
-      $("#ftw-showroom-prev", element).hide();
-    }
+    var nextButton = $("#ftw-showroom-next", element);
+    var prevButton = $("#ftw-showroom-prev", element);
+
+    current() < options.total ? nextButton.show() : nextButton.hide();
+    register.hasPrev() ? prevButton.show() : prevButton.hide();
```
Comment (bierik) [inline ftw/showroom/js/src/showroom.js]: Mmmh like it 👍

File: ftw/showroom/js/src/showroom.js [hunk 4]
```diff
@@ -146,21 +147,28 @@ module.exports = function Showroom(items = [], options) {
...
     items.map(item => $(item.element).on("click", select));
+    if (options.offset > 0) {
+      options.offset -= items.length;
+    }
     register.prepend(items);
+    checkArrows();
   }
 
-  function reset(items = []) {
+  function reset(items = [], offset = 0) {
```
Comment (bierik) [inline ftw/showroom/js/src/showroom.js]: how about giving the possibility to pass an options object as seen in the constructor. If we want to reset more parameters in the future we will be more flexible.

File: ftw/showroom/js/src/register.js (hunk 1 above)
Comment (deiferni) [inline ftw/showroom/js/src/register.js]: but this is a property which cannot be called (this is the register, not the showroom which also has a current method 😉 )

File: ftw/showroom/js/src/register.js (hunk 2 above)
Comment (deiferni) [inline ftw/showroom/js/src/register.js]: same here, this is a property which returns the current item (https://github.com/4teamwork/ftw.showroom/blob/master/ftw/showroom/js/src/register.js#L72)

File: ftw/showroom/js/src/showroom.js (hunk 4 above)
Comment (deiferni) [inline ftw/showroom/js/src/showroom.js]: mhh i change the code in the meantime to call `setOffset` with the passed-in offset, so i'd suggest to change the code when we need to reset more stuff, (if you agree).

Comment (bierik) [inline ftw/showroom/js/src/showroom.js]: 👍

File: ftw/showroom/js/src/register.js (hunk 2 above)
Comment (bierik) [inline ftw/showroom/js/src/register.js]: Oh I see nevermind

**Comment by deiferni** [comment]
@bierik updated with your suggestions

**Pull request merged**